src/main/java/io/github/moulberry/notenoughupdates/profileviewer/level/task/SlayingTaskLevel.java

```diff
@@ -0,0 +1,288 @@
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates;
+import io.github.moulberry.notenoughupdates.profileviewer.ExtraPage;
+import io.github.moulberry.notenoughupdates.profileviewer.GuiProfileViewer;
+import io.github.moulberry.notenoughupdates.profileviewer.ProfileViewer;
+import io.github.moulberry.notenoughupdates.profileviewer.level.LevelPage;
+import io.github.moulberry.notenoughupdates.util.Utils;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SlayingTaskLevel {
+
+	private final LevelPage levelPage;
+
+	public SlayingTaskLevel(LevelPage levelPage) {this.levelPage = levelPage;}
+
+	public void drawTask(JsonObject object, int mouseX, int mouseY, int guiLeft, int guiTop) {
+		// slayer
+		JsonObject slayingTask = levelPage.getConstant().get("slaying_task").getAsJsonObject();
+		JsonArray slayerLevelUpXp = slayingTask.get("slayer_level_up_xp").getAsJsonArray();
+		Map<String, ProfileViewer.Level> skyblockInfo = levelPage.getProfile().getSkyblockInfo(levelPage.getProfileId());
+		List<String> slayers = Arrays.asList("zombie", "spider", "wolf", "enderman", "blaze");
+
+		int sbXpGainedSlayer = 0;
+		if (skyblockInfo != null) {
+			for (String slayer : slayers) {
+				ProfileViewer.Level level = skyblockInfo.get(slayer);
+				for (int i = 0; i < (int) level.level; i++) {
+					int asInt = slayerLevelUpXp.get(i).getAsInt();
+					sbXpGainedSlayer += asInt;
+				}
+			}
+		}
+
+		JsonObject bossCollectionsXp = slayingTask.getAsJsonObject("boss_collections_xp");
+
+		HashMap<String, Double> allComps = new HashMap<>();
+		JsonElement normalCompletions = Utils
+			.getElement(object, "dungeons.dungeon_types.catacombs.tier_completions");
+
+		JsonElement masterCompletions = Utils
+			.getElement(object, "dungeons.dungeon_types.master_catacombs.tier_completions");
+
+		if (normalCompletions != null) {
+			HashMap<String, Double> normalCompMap = NotEnoughUpdates.INSTANCE.manager.gson.fromJson(
+				normalCompletions.getAsJsonObject(),
+				HashMap.class
+			);
+			normalCompMap.forEach((floor, value) -> {
+				if (allComps.containsKey(floor)) {
+					allComps.put(floor, allComps.get(floor) + value);
+				} else {
+					allComps.put(floor, value);
+				}
+			});
+		}
+		if (masterCompletions != null) {
+			HashMap<String, Double> masterCompMap = NotEnoughUpdates.INSTANCE.manager.gson.fromJson(
+				masterCompletions.getAsJsonObject(),
+				HashMap.class
+			);
+
+			masterCompMap.forEach((floor, value) -> {
+				if (allComps.containsKey(floor)) {
+					allComps.put(floor, allComps.get(floor) + value);
+				} else {
+					allComps.put(floor, value);
+				}
+			});
+		}
+		int[] bossLow = {25, 50, 100, 150, 250, 1000};
```

Comment:
These static helper arrays can be moved outside the method to class fields for less object creation